utils/oc.py

```diff
@@ -213,18 +214,64 @@ def recycle_pods(self, dry_run, namespace, dep_kind, dep_resource):
         pods = self.get(namespace, 'Pods')['items']
 
         if dep_kind == 'Secret':
-            pods_to_recycle = [pod['metadata']['name'] for pod in pods
+            pods_to_recycle = [pod for pod in pods
                                if self.secret_used_in_pod(dep_name, pod)]
         else:
             raise RecyclePodsUnsupportedKindError(dep_kind)
 
+        recyclables = {}
         for pod in pods_to_recycle:
-            logging.info(['recycle_pod', namespace, pod])
-            if not dry_run:
-                self.delete(namespace, 'Pod', pod)
-                logging.info(['validating_pods', namespace])
-                self.validate_pods_ready(
-                    namespace, self.secret_used_in_pod, dep_name)
+            if pod['status']['phase'] == 'Succeeded':
+                # don't try to recycle a pod that completed successfully
+                continue
+            owner = self.get_obj_root_owner(namespace, pod)
+            kind = owner['kind']
+            if kind == 'ReplicationController':
+                # need to act on the pod for replication controllers
+                owner = pod
+            recyclables.setdefault(kind, [])
+            exists = False
+            for obj in recyclables[kind]:
+                owner_name = owner['metadata']['name']
+                if obj['metadata']['name'] == owner_name:
+                    exists = True
+                    break
+            if not exists:
+                recyclables[kind].append(owner)
+
+        supported_recyclables = ['ReplicationController', 'Deployment',
+                                 'DeploymentConfig']
+        for kind, objs in recyclables.items():
+            for obj in objs:
+                if kind not in supported_recyclables:
```

Comment:
Should move this check outside of this loop.  If the kind isn't supported no point in looking at all objects.